File: github_javapop/app/src/main/java/com/lfernando/githubjavapop/adapter/RepositoryAdapter.java
package com.lfernando.githubjavapop.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.lfernando.githubjavapop.R;
import com.lfernando.githubjavapop.model.Repo;
import com.lfernando.githubjavapop.viewHolder.RepositoryViewHolder;
import com.squareup.picasso.Picasso;

import java.util.List;

public class RepositoryAdapter extends RecyclerView.Adapter {
    private List<Repo> repoList;
    private Context context;

    public RepositoryAdapter(List<Repo> repoList, Context context) {
        this.repoList = repoList;
        this.context = context;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(context).inflate(R.layout.repository_item, viewGroup, false);
        RepositoryViewHolder holder = new RepositoryViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int i) {
        RepositoryViewHolder holder = (RepositoryViewHolder) viewHolder;
        Repo repo = repoList.get(i);
        holder.repoName.setText(repo.getName());
        holder.repoDesc.setText(repo.getDescription());
        holder.forksCount.setText(String.valueOf(repo.getForks()));
        holder.starsCount.setText(String.valueOf(repo.getStars()));
        holder.name.setText(repo.getOwner().getLogin());
        holder.userName.setText(repo.getOwner().getName());
        Picasso.get().load(repo.getOwner().getAvatarUrl()).into(holder.avatar);

    }

    @Override
    public int getItemCount() {
        return repoList.size();
    }
}

File: app/src/main/java/matheusuehara/github/view/adapters/RepositoryAdapter.kt
package matheusuehara.github.view.adapters

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.lv_item_repository.view.*

import java.util.ArrayList
import matheusuehara.github.R
import matheusuehara.github.model.Repository
import matheusuehara.github.view.PullRequestActivity

class RepositoryAdapter(var repositories: ArrayList<Repository>, var context: Context) : RecyclerView.Adapter<RepositoryAdapter.ViewHolder>() {

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {
        var mRepositoryUserImage:ImageView = view.repository_user_image
        var mRepositoryName:TextView = view.repository_name
        var mRepositoryUserName:TextView = view.repository_username
        var mRepositoryFullName:TextView = view.repository_full_name
        var mRepositoryDescription:TextView = view.repository_description
        var mRepositoryStar:TextView = view.star_value
        var mRepositoryFork:TextView = view.fork_value

        init {
            view.setOnClickListener(this)
        }

        override fun onClick(v: View) {
            val position = adapterPosition
            try {
                val repository:Repository = repositories[position]
                val i = Intent(context, PullRequestActivity::class.java)
                i.putExtra("repositoryName", repository.name)
                i.putExtra("repositoryOwnerName", repository.owner.login)
                context.startActivity(i)
            } catch (e: ArrayIndexOutOfBoundsException) {
                e.printStackTrace()
            }
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view:View = LayoutInflater.from(parent.context)
                .inflate(R.layout.lv_item_repository, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val repository:Repository = repositories[position]
        holder.mRepositoryName.text = repository.name
        holder.mRepositoryDescription.text = repository.description
        holder.mRepositoryStar.text = repository.stargazers_count.toString()
        holder.mRepositoryFork.text = repository.forks_count.toString()
        holder.mRepositoryUserName.text = repository.owner.login
        holder.mRepositoryFullName.text = ""
        Picasso.get()
                .load(repository.owner.avatar_url)
                .error(R.mipmap.ic_launcher)
                .into(holder.mRepositoryUserImage)
    }

    override fun getItemCount(): Int {
        return repositories.size
    }
}